src/System.Buffers/src/System/Buffers/NativeBufferPool.cs

```diff
@@ -14,6 +14,23 @@ public unsafe sealed class NativeBufferPool<T> where T : struct
         private int _maxElements;
         private NativeBufferBucket<T>[] _buckets;
 
+        private static NativeBufferPool<byte> _sharedInstance = null;
+
+        public static NativeBufferPool<byte> SharedByteBufferPool
+        {
+            get
+            {
+                if (Volatile.Read(ref _sharedInstance) == null)
+                {
+                    // 2MB taken as the default since the average HTTP page is 1.9MB
+                    // per http://httparchive.org/interesting.php, as of October 2015
+                    Interlocked.CompareExchange(ref _sharedInstance, new NativeBufferPool<byte>(2048000), null);
```

Comment:
The pool isn't IDisposable so it doesn't need to have a Dispose call; also wouldn't it get GC'd if it was not exchanged since no variable would have a reference to it?